azure-pipelines.yml

```diff
@@ -467,6 +467,35 @@ jobs:
     parameters:
       includePatterns:
         - "/hw/***"
+  - task: DownloadSecureFile@1
+    name: gcpkey
+    inputs:
+      secureFile: gcpkey.json
+  - bash: |
```

Comment:
This is pretty complicated, so maybe it's worth pulling out into a separate script under ci/?